SingularityService/src/main/java/com/hubspot/singularity/scheduler/SingularityScheduler.java

```diff
@@ -787,9 +787,19 @@ private int numInstancesExpected(SingularityRequest request, SingularityPendingR
         nextInstanceNumber++;
       }
 
-      newTasks
-        .add(new SingularityPendingTask(new SingularityPendingTaskId(request.getId(), deployId, nextRunAt.get(), nextInstanceNumber, pendingRequest.getPendingType(), pendingRequest.getTimestamp()),
-          pendingRequest.getCmdLineArgsList(), pendingRequest.getUser(), pendingRequest.getRunId(), pendingRequest.getSkipHealthchecks(), pendingRequest.getMessage(), pendingRequest.getResources(), pendingRequest.getActionId()));
+      newTasks.add(
+          new SingularityPendingTask(
+              new SingularityPendingTaskId(
+                  request.getId(), deployId, nextRunAt.get(), nextInstanceNumber,
+                  pendingRequest.getPendingType(), pendingRequest.getTimestamp()),
+              pendingRequest.getCmdLineArgsList(),
+              pendingRequest.getUser(),
+              pendingRequest.getRunId(),
+              pendingRequest.getSkipHealthchecks(),
+              pendingRequest.getMessage(),
+              pendingRequest.getResources(),
+              null,
```

Comment:
this should be the last missing piece of the puzzle to connect it all the way through. The `SingularityPendingRequest` is the object that will carry the envOverrides from the RunNowRequest to the pending task.

In `RequestResource.scheduleImmediately` we create a `SingularityPendingRequest` from the provided `SingularityRunNowRequest`. That pending request will need to take in the environment overrides, so that we can later use them on this line to put them on the pending task.

The `RequestResource` calls `checkRunNowRequest` on the `SingularityValidator`, which is what actually constructs the object actually running some checks. That is where the envOverrides will need to be put on the `SingularityPendingRequest`